packages/lab/src/date-picker/DatePickerOverlayProvider.tsx

```diff
@@ -130,6 +134,10 @@ export const DatePickerOverlayProvider: React.FC<
       reason?: OpenChangeReason | undefined,
     ) => {
       if (newOpen) {
+        if (readOnly) {
```

Comment:
I haven't looked carefully but does this mean that providing defaultOpen and open will still open the overlay in read-only?

Reply:
it will but can pickup in my PR, which is testing this code.